### ISSUE TITLE
[JIT] enable normalization in autodiff

### ISSUE THREAD

**Issue by jjsjann123** [pytorch/pytorch#48091]
## 🚀 Feature

`batch_norm` & `layer_norm` should be supported by autodiff

## Motivation

Normalization provides fusion opportunities. A normalization accompanied with point-wise ops is a common pattern in training. In order for fusion to take place, we need normalization operations to be symbolically differentiable.

We have both layers disabled in autodiff right now:
https://github.com/pytorch/pytorch/blob/49f0e5dfeb64a928c6a2368dd5f86573b07d20fb/torch/csrc/jit/runtime/symbolic_script.cpp#L992
https://github.com/pytorch/pytorch/blob/49f0e5dfeb64a928c6a2368dd5f86573b07d20fb/torch/csrc/jit/runtime/symbolic_script.cpp#L1016-L1017

cc @gmagogsfm

**Comment by jjsjann123** [comment]
Adding more to this topic, we should avoid in-place batchnorm in differentiable graph for the sake of codgen. (There has been many long discussion on our buggy batchnorm schema, here's a better one discussing the problem: https://github.com/pytorch/pytorch/issues/13402)

Batchnorm updates running stats in-place, this raises challenge to codegen as in-place operation complicates dependency analysis.

Can we transform Batchnorm to a non-in-place version in autodiff? Would that raise issue for double backward?

**Comment by eellison** [comment]
@jjsjann123 are you working on this ?

**Comment by jjsjann123** [comment]
Almost forgot this ancient issue.

Yes, I still am. Currently working on BN in our internal codegen branch, I'll upstream JIT related changes when it's in a working condition.
Meanwhile, for layernorm, I do have an inflight PR that is good for review (I need to resolve the conflicts) #50467

**Comment by davidberard98** [comment]
@jjsjann123 this is fixed by #50467 and #57321 right? can we close it?
